### PR TITLE
Add some AddressBook and Carbon stubs

### PR DESCRIPTION
**Pull request by kamillys**

### CODE REVIEW

**Comment by ahyattdev** [comment]
Nice

File: src/Carbon/include/HIToolbox/TextInputSources.h [hunk 1]
```diff
@@ -0,0 +1,19 @@
+#ifndef _Carbon_TextInputSources_H_
+#define _Carbon_TextInputSources_H_
+
+#ifndef __cplusplus
```
Comment (CuriousTommy) [inline src/Carbon/include/HIToolbox/TextInputSources.h]: I don't have any experience with `extern "C"`, but going off of [stack overflow](https://stackoverflow.com/a/16192947/5988706), it should be `#ifdef __cplusplus` not `#ifndef __cplusplus`

However, I haven't tested this, so I could be wrong.

File: src/Carbon/include/HIToolbox/TextInputSources.h [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+#endif
+
+extern const CFStringRef kTISPropertyInputSourceID;
+extern const CFStringRef kTISPropertyUnicodeKeyLayoutData;
+extern const CFStringRef kTISCategoryKeyboardInputSource;
+extern const CFStringRef kTISPropertyInputSourceCategory;
+extern const CFStringRef kTISPropertyInputSourceType;
+extern const CFStringRef kTISTypeKeyboardLayout;
+
+#ifndef __cplusplus
```
Comment (CuriousTommy) [inline src/Carbon/include/HIToolbox/TextInputSources.h]: Same goes here.

File: src/Carbon/include/HIToolbox/TextInputSources.h (hunk 1 above)
Comment (CuriousTommy) [inline src/Carbon/include/HIToolbox/TextInputSources.h]: I don't have any experience with `extern "C"`, but going off of [stack overflow](https://stackoverflow.com/a/16192947/5988706), it should be `#ifdef __cplusplus` not `#ifndef __cplusplus`

However, I haven't tested this, so I could be wrong.

File: src/Carbon/include/HIToolbox/TextInputSources.h (hunk 2 above)
Comment (CuriousTommy) [inline src/Carbon/include/HIToolbox/TextInputSources.h]: Same goes here.

File: src/Carbon/include/HIToolbox/TextInputSources.h (hunk 1 above)
Comment (kamillys) [reply src/Carbon/include/HIToolbox/TextInputSources.h]: True, my bad.

**Pull request merged**